src/main/java/nebula/plugin/metrics/collector/GradleCollector.java

```diff
@@ -171,24 +175,27 @@ public void buildFinished(BuildProfile result) {
      * Conditionally shutdown the dispatcher, because Gradle listener event order appears to be non-deterministic.
      */
     private void shutdownIfComplete() {
+        // only shut down if you have updated build results AND profile information
+        if (!buildProfileComplete.get() || !buildResultComplete.get()) {
```

Comment:
Is this what was causing the problems we were seeing, or is this something new because of the refactored behaviour of the clients?